File: src/main/scala/actors5/Escritor.scala
package hello.actors5

import scala.actors.Actor
import scala.actors.Actor._
import java.io._
import java.net.Socket

object Escritor extends Actor {

  def act() {
    loop {
      react {
        case s:Socket =>
          Lector ! s
        case Saluda(nom,cuando,socket) =>
          println("Nos enviaron '" + nom + "' a las " + cuando + " (ahorita son las " + new java.util.Date + ")")
          val out = new PrintStream(socket.getOutputStream)
          out.printf("Hola, %s!%n", nom)
          out.flush()
          socket.close()
      }
    }
  }
}

File: src/main/java/hola/Escritor.java
package hola;
import java.net.Socket;
import java.io.*;

public class Escritor {

  public static void escribe(String nombre, Socket sock) throws IOException {
    PrintStream pout = new PrintStream(sock.getOutputStream());
    pout.printf("Hola, %s!%n", nombre);
    pout.flush();
    sock.close();
  }

}
